zerver/tests/test_decorators.py

```diff
@@ -2079,7 +2079,11 @@ def inner(request: HttpRequest) -> HttpResponse:
             return inner
 
         zulip_realm = get_realm("zulip")
-        with mock.patch("zerver.views.home.home_real", new=mock_home(zulip_realm)):
+
+        # We don't need to test if user is logged in here, so we patch zulip_login_required.
+        with mock.patch("zerver.views.home.home_real", new=mock_home(zulip_realm)) and mock.patch(
+            "zerver.decorator.zulip_login_required", lambda f: f
```

Comment:
@amanagr that's very likely a bug with the test; let's discuss it in the testing stream and probably @mateuszmandera will be able to help debug.

Reply:
@amanagr @timabbott  Shouldn't we just be patching zulip_login_required in `home`?
```
        with mock.patch("zerver.views.home.home_real", new=mock_home(zulip_realm)) and mock.patch(
            "zerver.views.home.zulip_login_required", lambda f: f
        ):
```

Though another problem pops up, becase then this starts failing until the mock is added:
```
        # When a request is made to the root subdomain and there is no realm on it,
        # no realm can be set on the request notes.
        with mock.patch("zerver.views.home.home_real", new=mock_home(None)):
            result = self.client_get("/", subdomain="")
            self.assertEqual(result.status_code, 200)
```

We can fix it by adding the decorator mock again as well. Which then leads to failure because we get `InvalidSubdomainError` from `get_valid_realm_from_request`. I think we can get around it by just adding a bunch of more annoying mocking to prevent failures. Or it might be better to just rewrite this step to not query `/` and just query `/new/` instead? Because /new/ is an actual endpoint that is meant to be accessible on the root subdomain.

Reply:
Yeah, I tried to do `zulip_login_required in home` first but it was not working as you mentioned. Thanks for helping!